Source/WebCore/workers/service/server/SWServerWorker.cpp

```diff
@@ -60,7 +60,7 @@ SWServerWorker::SWServerWorker(SWServer& server, SWServerRegistration& registrat
     , m_contentSecurityPolicy(contentSecurityPolicy)
     , m_crossOriginEmbedderPolicy(crossOriginEmbedderPolicy)
     , m_referrerPolicy(WTFMove(referrerPolicy))
-    , m_registrableDomain(m_data.scriptURL)
+    , m_registrableDomain(RegistrableDomain::uncheckedCreateFromHost(m_registrationKey.topOrigin().host))
```

Comment:
I am not sure this is correct. 
See for instance line 67 below:
m_lastNavigationWasAppInitiated(m_server->clientIsAppInitiatedForRegistrableDomain(m_registrableDomain))

We might want to introduce a registrableDomainForCookies or something like that, which could use m_registrationKey to get the top level origin domain.